pages/forms/TripForm/[id].js

```diff
@@ -13,16 +13,16 @@ export default function TripDetailsForm() {
   const [endDateDisabled, setEndDateDisabled] = useState(false);
 
   const { id } = router.query;
-  const { data: trips, isLoading } = useSWR(
+  const { data: trip, isLoading } = useSWR(
     id ? `/api/trips/${id}` : null,
     fetcher
   );
 
-  if (!trips || isLoading) {
+  if (!trip || isLoading) {
     return <h2>is Loading...</h2>;
   }
 
-  const { title, location, timePeriod, img } = trips;
+  const { title, location, timePeriod, img } = trip;
   const country = location.map((location) => `${location.country}`);
   const city = location.map((location) => `${location.city}`);
   const startDate = timePeriod.map((timePeriod) => `${timePeriod.startDate}`);
```

Comment:
While I still don't think location and time information should be stored in arrays (you just have one of either!), you can at least get rid of the map:

```js
const { country, city } = location[0];
const { startDate, endDate } = timePeriod[0];
```

Here for instance:

```js
<input name="country" defaultValue={country} minLength={3} required />;
```

You are passing an array (`country`) as the `defaultValue` of the input. It works just because you have a single entry inside your array, and its string representation corresponds to that single entry as a string, but ideally you shouldn't be able to pass an array to a single input.

To be clear, I absolutely agree with having `{ location: { country: '...', city: '...' } }`, it's just the `[...]` around that are redundant ^^

Reply:
Hi Diego, now we fully understand it. We got rid of the map (for now), but just in this file and we we will refactor the schema and then the mapping in an own user story! 

Then, we don't need this anymore, but it helped us to understand what you mean ✌️
`const { country, city } = location[0];
const { startDate, endDate } = timePeriod[0];`